### PR TITLE
fixes get osd_id from directory name

### LINKED ISSUE #579: osd_directory daemon do not work with CLUSTER names with "-"
The image allow to set the ceph cluster name via the CLUSTER environment variable,
but the osd_directory daemon do not works with names that includes "-" like "kube-system" for example.

with cluster names like "kube-system" the daemon will use "system" as OSD_ID

The error seems comes from:
https://github.com/ceph/ceph-docker/blob/master/ceph-releases/kraken/ubuntu/16.04/daemon/osd_scenarios/osd_directory.sh#L39

### PR DESCRIPTION
**Pull request by osixia**
get the OSD_ID from directory name after the last "-"

for example:
kube-system-3 -> OSD_ID = "3"

instead of OSD_ID="system" with the current command
fixes #579

### CODE REVIEW

**Comment by leseb** [comment]
jenkins test ceph_ansible-kraken-centos7_cluster

File: ceph-releases/kraken/ubuntu/16.04/daemon/watch_mon_health.sh [hunk 1]
```diff
@@ -2,7 +2,6 @@
 set -e
 
 function watch_mon_health {
-  log "checking for zombie mons"
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/watch_mon_health.sh]: why?

File: ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py [hunk 2]
```diff
@@ -11,19 +11,18 @@
...
     kubectl_command = 'kubectl get pods --namespace=${CLUSTER} -l daemon=mon -o template --template="{ {{range  \$i, \$v  := .items}} {{ if \$i}} , {{ end }} \\"{{\$v.metadata.name}}\\": \\"{{\$v.status.podIP}}\\" {{end}} }"'
 
-monmap_command = "ceph mon getmap > /tmp/monmap && monmaptool -f /tmp/monmap --print"
+monmap_command = "ceph --cluster=${CLUSTER} mon getmap > /tmp/monmap && monmaptool -f /tmp/monmap --print"
 
 
 def extract_mons_from_monmap():
-    monmap = subprocess.check_output(monmap_command, shell=True)
-    mons = {}
+    monmap =  subprocess.check_output(monmap_command, shell=True)
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py]: extra space?

File: ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py [hunk 3]
```diff
@@ -11,19 +11,18 @@
...
+    mons = { }
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py]: why?

**Comment by osixia** [comment]
oops sorry i'm not familiar with github pull requests and it seems that the commit that try to fix #581 as been added to this one :(

File: ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py [hunk 4]
```diff
@@ -36,13 +35,13 @@ def extract_mons_from_kubeapi():
...
-        subprocess.call(["ceph", "mon", "remove", mon])
+    if not mon in expected_mons:
+        print "removing zombie mon ", mon
+        subprocess.call(["ceph", "--cluster", os.environ["CLUSTER"], "mon", "remove", mon])
         removed_mon = True
-    elif current_mons[mon] != expected_mons[mon]:  # check if for some reason the ip of the mon changed
+    elif current_mons[mon] != expected_mons[mon]: # check if for some reason the ip of the mon changed
         print "ip change dedected for pod ", mon
-        subprocess.call(["kubectl", "--namespace=ceph", "delete", "pod", mon])
+        subprocess.call(["kubectl", "--namespace", os.environ["CLUSTER"], "delete", "pod", mon])
```
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py]: does that work without the `=`?

File: ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py (hunk 3 above)
Comment (osixia) [inline ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py]: my editor may have try to "beautify" the code, no particular reasons

File: ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py (hunk 4 above)
Comment (osixia) [inline ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py]: yes :)
```
kubectl --namespace osixia-system get pods 
NAME                              READY     STATUS             RESTARTS   AGE
ceph-mon-0ww8j                    1/1       Running            67         16h
ceph-mon-2xqr9                    1/1       Running            1          14h
ceph-mon-check-2024006442-vvvbs   1/1       Running            0          16m
ceph-osd-9bm5l                    1/1       Running            0          15h
ceph-osd-kz3sd                    1/1       Running            0          15h
ceph-osd-p5hxq                    1/1       Running            69         15h
```

File: ceph-releases/kraken/ubuntu/16.04/daemon/watch_mon_health.sh (hunk 1 above)
Comment (osixia) [inline ceph-releases/kraken/ubuntu/16.04/daemon/watch_mon_health.sh]: this is also log just after in the while loop so at first it print it twice:
2017-03-30 07:44:57  /entrypoint.sh: checking for zombie mons
2017-03-30 07:44:57  /entrypoint.sh: checking for zombie mons

and while instruction can't fail.

Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/watch_mon_health.sh]: Ack, thanks.

File: ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py (hunk 3 above)
Comment (leseb) [inline ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py]: But this is incorrect :). E201 from pep8. Please revert this.

Comment (osixia) [inline ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py]: must be done :)
this was due to a copy past from an other check_zombie_mons.py so i also change it in the other files

**Comment by leseb** [comment]
Please run the following commands locally:

* first rebase against master: `git pull origin master --rebase`
* then rebase using: `git rebase -i e20591c43e77708082462b6392c2461a277d38e8`
* Modify the lines to end up with:

```
pick 15032fc fixes get osd_id from directory name
pick c86371d make check_zombie_mons.py works cluster name
s 52859d6 revert formatting changes
s d1a8250 { } -> {} and remove extra space
```

* then squash your commit to merge your last three by removing some lines, you should end up with:

```
make check_zombie_mons.py works cluster name

# Please enter the commit message for your changes. Lines starting
# with '#' will be ignored, and an empty message aborts the commit.
#
# Author:    Bertrand Gouny <bertrand.gouny@osixia.net>
# Date:      Thu Mar 30 10:24:15 2017 +0200
#
# interactive rebase in progress; onto e20591c
# Last commands done (4 commands done):
#    s 52859d6 revert formatting changes
#    s d1a8250 { } -> {} and remove extra space
# No commands remaining.
# You are currently editing a commit while rebasing branch 'osixia-master' on 'e20591c'.
#
# Changes to be committed:
#       modified:   ceph-releases/hammer/centos/7/daemon/check_zombie_mons.py
#       modified:   ceph-releases/infernalis/ubuntu/14.04/daemon/check_zombie_mons.py
#       modified:   ceph-releases/infernalis/ubuntu/14.04/daemon/entrypoint.sh
#       modified:   ceph-releases/jewel/ubuntu/14.04/daemon/check_zombie_mons.py
#       modified:   ceph-releases/jewel/ubuntu/14.04/daemon/entrypoint.sh
#       modified:   ceph-releases/kraken/ubuntu/16.04/daemon/check_zombie_mons.py
#       modified:   ceph-releases/kraken/ubuntu/16.04/daemon/watch_mon_health.sh
#
# ------------------------ >8 ------------------------
# Do not touch the line above.
# Everything below will be removed.
```

* check your git log, you should have only 2 commits
* finally git push force using the `-f` when pushing your changes.

**Comment by osixia** [comment]
not sure to understand but i have an error with git rebase:

git rebase -i e20591c43e77708082462b6392c2461a277d38e8 
```
fatal: Needed a single revision
invalide e20591c43e77708082462b6392c2461a277d38e8 en amont
```

**Comment by leseb** [comment]
Please run:

`git remote add ceph-docker git@github.com:ceph/ceph-docker.git``

Then run `git pull ceph-docker master --rebase`.
Then resume at `git rebase -i e20591c`.

**Comment by osixia** [comment]
is that ok ?

**Comment by leseb** [comment]
@osixia Super merci !

**Pull request merged**